src/read.c

```diff
@@ -4114,14 +4102,18 @@ avifResult avifDecoderNthImage(avifDecoder * decoder, uint32_t frameIndex)
         }
         // The next image (decoder->imageIndex + 1) is partially decoded but
         // the previous image (decoder->imageIndex) is requested.
-        // Fall through to flush and start decoding from the nearest key frame.
+        // Fall through to resetting the decoder data and start decoding from
+        // the nearest key frame.
     }
 
     int nearestKeyFrame = (int)avifDecoderNearestKeyframe(decoder, frameIndex);
     if ((nearestKeyFrame > (decoder->imageIndex + 1)) || (requestedIndex <= decoder->imageIndex)) {
-        // If we get here, a decoder flush is necessary
+        // If we get here, we need to start decoding from the nearest key frame.
+        // So discard the unused decoder state and its previous frames. This
+        // will force the setup of a new decoder instance in
```

Comment:
Nit: Please clarify "a new decoder instance". It is ambiguous because "decoder" could be misinterpreted as avifDecoder. But the comment is referring to avifCodec instances.

One way to clarify this is to change "a new decoder instance" to "new AV1 decoder instances". Note the plural, because we may create multiple AV1 decoder instances, e.g., to decode the color and alpha sub-images.

An alternative is to say "new avifCodec instances".

Reply:
Oops, i just merged the change. I will update the comment in a follow-up PR.

Reply:
done here: https://github.com/AOMediaCodec/libavif/pull/1317